sky/resources.py

```diff
@@ -85,9 +85,14 @@ def __init__(
 
         self._image_id = image_id
         if isinstance(image_id, str):
-            self._image_id = {self._region: image_id}
-        elif isinstance(image_id, dict) and None in image_id:
-            self._image_id = {self._region: image_id[None]}
+            self._image_id = {self._region: image_id.strip()}
+        elif isinstance(image_id, dict):
+            if None in image_id:
+                self._image_id = {self._region: image_id[None].strip()}
+            else:
+                self._image_id = {
+                    k.strip(): v.strip() for k, v in image_id.items()
+                }
```

Comment:
Why do these image_ids need strip()?

Reply:
It seems when we add a space to the image_id, the launching will fail and can be quite confusing.